jetty-client/src/main/java/org/eclipse/jetty/client/HttpDestination.java

```diff
@@ -530,16 +577,38 @@ public String asString()
         return origin.asString();
     }
 
+    private long getStaleDelayBeforeExpiration()
+    {
+        if (idleDestinationsSweeper == null)
+            return -1;
+        try (Locker.Lock l = staleLock.lock())
+        {
+            long destinationIdleTimeout = TimeUnit.MILLISECONDS.toNanos(getHttpClient().getDestinationIdleTimeout());
+            return TimeUnit.NANOSECONDS.toMillis(destinationIdleTimeout - (System.nanoTime() - this.staleTs));
+        }
+    }
+
+    private boolean isStale()
+    {
+        try (Locker.Lock l = staleLock.lock())
+        {
+            return this.stale;
+        }
+    }
+
     @Override
     public String toString()
     {
-        return String.format("%s[%s]@%x%s,queue=%d,pool=%s",
+        return String.format("%s[%s]@%x%s,state=%s,queue=%d,pool=%s,stale=%s,staleDelay=%d",
```

Comment:
Don't like the word `delay`, as it's not really a delay.
How about just `stale=%b,idle=%d` where `idle` is filled with `System.nanoTime() - staleTs` in ms, which then can also be exposed via JMX?

Reply:
Done.